Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -177,6 +175,37 @@ public AsyncFunctionAssertions(Func<TTask> subject, IExtractExceptions extractor
         return ThrowInternal<TException>(exception, because, becauseArgs);
     }
 
+    /// <summary>
+    /// Asserts that the current <see cref="Func{Task}"/> throws an exception of type <typeparamref name="TException"/>.
+    /// </summary>
+    /// <typeparam name="TException">The type of exception expected to be thrown.</typeparam>
+    /// <param name="timeSpan">The allowed time span for the operation.</param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public async Task<ExceptionAssertions<TException>> ThrowWithinAsync<TException>(
+        TimeSpan timeSpan, string because = "", params object[] becauseArgs)
+        where TException : Exception
+    {
+        bool success = Execute.Assertion
+            .ForCondition(Subject is not null)
+            .BecauseOf(because, becauseArgs)
+            .FailWith("Expected {context} to throw {0} within {1}{reason}, but found <null>.",
+                typeof(TException), timeSpan);
+
+        if (!success)
```

Comment:
Let's follow the project style of not using early returns.
(despite our shared personal preference 😉)

Reply:
> Let's follow the project style of not using early returns.

Early returns at the top of the method are fine by me too. Just not the ones hidden in the middle of the code base.

Reply:
Mmh, I just copied from Jonas' changes above...
I'll change it in the same way when discussion in #2105 is completed.

Reply:
done